### PR TITLE
Recompile on Edit [RIP67]

### PR DESCRIPTION
**Pull request by AlexWeinstein92**
- error highlights on right side of editor show up when errors are typed
  - still need to remove the highlights (RIP41)
- `parseRootWithURLs` in Settings on apply
- `parseNebulaContents` in `runCommandForEditor` via `DocumentListener.documentChanged`
- various bugs fixed from last PR
- uses RIDDL 0.55.0 (unpublished, local build)

### CODE REVIEW

File: build.sbt
Comment (reid-spencer) [inline build.sbt:21]: It's not a good idea to put that kind of version number in a build. At the very least, commit your changes and use the new commit number, like 0.55.0-2-ab83e423. The hash, ab83e423, for the commit is fictitious so use the one that you get when you commit your changes. 

The reason it is a bad idea is because it precludes team work. NOBODY but you on your machine could recreate the same version of RIDDL. but, if you commit your work, they can.

File: project/Dependencies.scala
Comment (reid-spencer) [inline project/Dependencies.scala:27]: Same comment as above for the Wit.riddl version

File: src/main/scala/com/ossuminc/riddl/plugins/idea/files/RiddlDocumentListener.scala
Comment (reid-spencer) [inline src/main/scala/com/ossuminc/riddl/plugins/idea/files/RiddlDocumentListener.scala:39]: THere's a replacement for this on the head of main branch now. You can tokenized the input using the TopLevelParser. It won't parse for correctness, just tokenized so should handle misspellings and bad syntax with ease.

**Review by reid-spencer — CHANGES_REQUESTED** [review]
This is too long to review in detail at 5pm but I made some comments about your build that you should change. In general, the immense amount of logic you're dealing with for highlighting could probably be changed to a TopLevelParser.parseToTokens call and a match statement on each of the resulting tokens. 

It helps my review greatly if you can put a video showing the visual results.

File: build.sbt
Comment (reid-spencer) [reply build.sbt:21]: The dynver.sbt plugin generates that version number. It does so by starting with a tag (0.55.0) appending how many commits past that you are at (1) and the commit hash of that commit (c184a0c5). If you’ve also made local changes, uncommitted, it appends the latest timestamp of the modified files (20241121-115, i.e. 1:15 am UTC last Thursday).
So, from that I surmised that you MUST have a local change. Had it just been 0.55.0-1-c185a0c5, I would not have made that comment because that’s the kind of version number generated when no changes have been made.

File: build.sbt
Comment (reid-spencer) [inline build.sbt:21]: Better! But you might want to try 0.56.0 now?

File: project/Dependencies.scala
Comment (reid-spencer) [inline project/Dependencies.scala:27]: 0.56.0

**Review by reid-spencer — APPROVED** [review]
LGTM, please make sure it compiles with 0.56.0 before merging.

**Comment by AlexWeinstein92** [comment]
Note: removing highlights & their hover text is part of RIP-41
https://github.com/user-attachments/assets/91000e0a-068b-43f9-b21c-812e06feef6f

**Comment by AlexWeinstein92** [comment]
Note: removing highlights & their hover text is part of RIP-41

https://github.com/user-attachments/assets/5128f47a-1231-45d1-84e9-3261ac4733b5

**Review by reid-spencer — APPROVED** [review]
The highlighting looks MUCH better, even through editing. I will approve this but it would be better if the color change was as I suggested last week. That is the keywords need to be orange and the "readability" words need to be purple, and the punctuation needs to be red.  If in doubt, look at the way GitHub shows scala code and that's what we're going for except we highlight strings, comments and markdown blocks too.

**Pull request merged**
